zavod/zavod/exporters/peps.py

```diff
@@ -0,0 +1,111 @@
+from collections import defaultdict
+from typing import Dict, List, Any, Optional, Set, DefaultDict, NewType
+from followthemoney import model
+
+from zavod.entity import Entity
+from zavod.exporters.common import Exporter
+from zavod.util import write_json
+from zavod import helpers as h
+
+
+CountryCode = NewType("CountryCode", str)
+PositionLabel = NewType("PositionLabel", str)
+PositionId = NewType("PositionId", str)
+
+# country code -> position id -> position summary
+PositionCount = Dict[str, str | int]
+PositionSummary = DefaultDict[PositionId, PositionCount]
+Country = Dict[str, PositionSummary | str | int]
+CountryMap = DefaultDict[CountryCode, Country]
+
+# position -> country code -> country summary
+CountryCount = Dict[str, str | int]
+CountrySummary = DefaultDict[CountryCode, PositionCount]
+Position = Dict[str, CountrySummary | str | int]
+PositionMap = DefaultDict[str, Position]
+
+
+class PEPSummaryExporter(Exporter):
+    TITLE = "PEP position occupancy summary"
+    FILE_NAME = "pep-positions.json"
+    MIME_TYPE = "application/json"
+
+    def setup(self) -> None:
+        super().setup()
+
+        self.countries: DefaultDict[str, Any] = defaultdict(
+            lambda: {
+                "positions": defaultdict(
+                    lambda: {
+                        "position_name": "",
+                        "counts": {
+                            "total": 0,
+                            h.Status.CURRENT.value: 0,
+                            h.Status.ENDED.value: 0,
+                            h.Status.UNKNOWN.value: 0,
+                        },
+                    }
+                ),
+                "counts": {
+                    "total": 0,
+                    h.Status.CURRENT.value: 0,
+                    h.Status.ENDED.value: 0,
+                    h.Status.UNKNOWN.value: 0,
+                },
+            }
+        )
+        self.positions: DefaultDict[str, Any] = defaultdict(
+            lambda: {
+                "countries": defaultdict(
+                    lambda: {
+                        "counts": {
+                            "total": 0,
+                            h.Status.CURRENT.value: 0,
+                            h.Status.ENDED.value: 0,
+                            h.Status.UNKNOWN.value: 0,
+                        }
+                    }
+                ),
+                "counts": {
+                    "total": 0,
+                    h.Status.CURRENT.value: 0,
+                    h.Status.ENDED.value: 0,
+                    h.Status.UNKNOWN.value: 0,
+                },
+            }
+        )
+
+    def observe_occupancy(self, occupancy: Entity, position: Entity) -> None:
+        country_codes = position.get("country")
+        for code in country_codes:
```

Comment:
I was having trouble emitting two countries, but I think it would be nice if we could emit both countries for a position that clearly spans countries like diplomats. It makes the counts a bit funny but we could add a footnone "Positions spanning multiple countries are counted in each country"